honeybee_energy/material/_base.py

```diff
@@ -0,0 +1,91 @@
+# coding=utf-8
+"""Base energy material."""
+from __future__ import division
+
+from honeybee.typing import valid_ep_string
+
+
+class _EnergyMaterialBase(object):
+    """Base energy material.
+
+    Properties:
+        name
+    """
+    _name_descr = 'material name'
+    _is_window_material = None
+
+    def __init__(self, name):
+        """Initialize energy material base.
+
+        Args:
+            name: Text string for material name. Must be <= 100 characters.
+                Can include spaces but special characters will be stripped out.
+        """
+        self.name = name
+
+    @property
+    def name(self):
+        """Get or set the text string for material name."""
+        return self._name
+
+    @name.setter
+    def name(self, name):
+        self._name = valid_ep_string(name, self._name_descr)
```

Comment:
why not just hard code `'material name'` here? I assume you don't overwrite this method in subclasses so you probably won't reuse `self._name_descr`.

Reply:
I originally had it that way but I changed it for 2 reasons (one of which is a good one and the other I am not sure about):

1) There are some cases where I reuse the input name since we have some materials with 30 inputs (like blinds) and I don't need to give a different input name for `back_beam_solar_reflectance` and `front_diffuse_solar_reflectcance` (Just `solar_reflectance` is fine). This is the good reason but it doesn't really apply to the `_name_descr` here.

2) The way I put the `_name_descr` variable, I know that it will use the same string for every instance of the class, thereby saving some memory from the case where every class has a new string.  My understanding was that hard-coding a string here will make a new string for each and every instance of the material but I have to admit I was not entirely sure of this. Do you know if my assumption is at all correct?